### PR TITLE
CGSN Gross Range & Climatology Tables for fixed-depth and profiling DOSTAs

### PR DESCRIPTION
**Pull request by reedan88**

### CODE REVIEW

File: qartod/dosta/climatology_tables/GA01SUMO-RII11-02-DOSTAD032-dissolved_oxygen.csv [hunk 1]
```diff
@@ -0,0 +1 @@
+,
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GA01SUMO-RII11-02-DOSTAD032-dissolved_oxygen.csv:1]: This file is empty. Is that correct?

File: qartod/dosta/climatology_tables/GA01SUMO-RII11-02-DOSTAD033-dissolved_oxygen.csv [hunk 2]
```diff
@@ -0,0 +1 @@
+,
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GA01SUMO-RII11-02-DOSTAD033-dissolved_oxygen.csv:1]: This file is empty. Is that correct?

File: qartod/dosta/climatology_tables/GA01SUMO-SBD11-04-DOSTAD000-dissolved_oxygen.csv [hunk 3]
```diff
@@ -0,0 +1,2 @@
+,"[1, 1]","[2, 2]","[3, 3]","[4, 4]","[5, 5]","[6, 6]","[7, 7]","[8, 8]","[9, 9]","[10, 10]","[11, 11]","[12, 12]"
+"[0, 0]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GA01SUMO-SBD11-04-DOSTAD000-dissolved_oxygen.csv:2]: All NaNs

File: qartod/dosta/climatology_tables/GA02HYPM-WFP02-03-DOSTAL000-dissolved_oxygen.csv [hunk 4]
```diff
@@ -0,0 +1,73 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GA02HYPM-WFP02-03-DOSTAL000-dissolved_oxygen.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GA01SUMO-RII11-02-DOSTAD033-dissolved_oxygen.csv (hunk 2 above)
Comment (reedan88) [reply qartod/dosta/climatology_tables/GA01SUMO-RII11-02-DOSTAD033-dissolved_oxygen.csv:1]: Probably not. I'll fix that.

File: qartod/dosta/climatology_tables/GA02HYPM-WFP02-03-DOSTAL000-estimated_oxygen_concentration.csv [hunk 5]
```diff
@@ -0,0 +1,73 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GA02HYPM-WFP02-03-DOSTAL000-estimated_oxygen_concentration.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GA01SUMO-RII11-02-DOSTAD032-dissolved_oxygen.csv (hunk 1 above)
Comment (reedan88) [reply qartod/dosta/climatology_tables/GA01SUMO-RII11-02-DOSTAD032-dissolved_oxygen.csv:1]: Probably not. I'll fix that.

File: qartod/dosta/climatology_tables/GA02HYPM-WFP03-03-DOSTAL000-dissolved_oxygen.csv [hunk 6]
```diff
@@ -0,0 +1,98 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GA02HYPM-WFP03-03-DOSTAL000-dissolved_oxygen.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GA02HYPM-WFP03-03-DOSTAL000-estimated_oxygen_concentration.csv [hunk 7]
```diff
@@ -0,0 +1,98 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GA02HYPM-WFP03-03-DOSTAL000-estimated_oxygen_concentration.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GI02HYPM-WFP02-03-DOSTAL000-dissolved_oxygen.csv [hunk 8]
```diff
@@ -0,0 +1,73 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GI02HYPM-WFP02-03-DOSTAL000-dissolved_oxygen.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GI02HYPM-WFP02-03-DOSTAL000-estimated_oxygen_concentration.csv [hunk 9]
```diff
@@ -0,0 +1,73 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GI02HYPM-WFP02-03-DOSTAL000-estimated_oxygen_concentration.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GP02HYPM-WFP02-03-DOSTAL000-dissolved_oxygen.csv [hunk 10]
```diff
@@ -0,0 +1,69 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GP02HYPM-WFP02-03-DOSTAL000-dissolved_oxygen.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GP02HYPM-WFP02-03-DOSTAL000-estimated_oxygen_concentration.csv [hunk 11]
```diff
@@ -0,0 +1,69 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GP02HYPM-WFP02-03-DOSTAL000-estimated_oxygen_concentration.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GP02HYPM-WFP03-03-DOSTAL000-dissolved_oxygen.csv [hunk 12]
```diff
@@ -0,0 +1,88 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GP02HYPM-WFP03-03-DOSTAL000-dissolved_oxygen.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GP02HYPM-WFP03-03-DOSTAL000-estimated_oxygen_concentration.csv [hunk 13]
```diff
@@ -0,0 +1,88 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GP02HYPM-WFP03-03-DOSTAL000-estimated_oxygen_concentration.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GS02HYPM-WFP02-03-DOSTAL000-dissolved_oxygen.csv [hunk 14]
```diff
@@ -0,0 +1,102 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[-59.7638, 168.7589]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GS02HYPM-WFP02-03-DOSTAL000-dissolved_oxygen.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GS02HYPM-WFP02-03-DOSTAL000-estimated_oxygen_concentration.csv [hunk 15]
```diff
@@ -0,0 +1,102 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[173.7586, 395.7965]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GS02HYPM-WFP02-03-DOSTAL000-estimated_oxygen_concentration.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GS02HYPM-WFP03-03-DOSTAL000-dissolved_oxygen.csv [hunk 16]
```diff
@@ -0,0 +1,91 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GS02HYPM-WFP03-03-DOSTAL000-dissolved_oxygen.csv:2]: Lots of NaNs, suggest deleting these rows

File: qartod/dosta/climatology_tables/GS02HYPM-WFP03-03-DOSTAL000-estimated_oxygen_concentration.csv [hunk 17]
```diff
@@ -0,0 +1,91 @@
+,"[1,1]","[2,2]","[3,3]","[4,4]","[5,5]","[6,6]","[7,7]","[8,8]","[9,9]","[10,10]","[11,11]","[12,12]"
+"[0, 5]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]","[nan, nan]"
```
Comment (cwingard) [inline qartod/dosta/climatology_tables/GS02HYPM-WFP03-03-DOSTAL000-estimated_oxygen_concentration.csv:2]: Lots of NaNs, suggest deleting these rows

**Review by cwingard — APPROVED** [review]
Multiple climatology tables have NaN's. Fairly certain these cannot be fed into the QARTOD code, but I could be mistaken. In multiple cases you just need to remove the depth bins that the profiler doesn't cover and never can given the way it is deployed. Also, multiple tables are essentially duplicates. The variables `estimated_oxygen_concentration` and `dosta_abcdjm_cspp_tc_oxygen` are the same. One is calculated onboard the sensor and the other is calculated onshore by CI. Unless there was an error in where/how the calibration coefficients were loaded, they will essentially identical (in fact, that is a good QC check). Those two sets of tables should be combined into a single table that the two variables can reference. That will make reviewing and managing these values easier long term.

**Pull request closed without merging**